device/alpaka/src/utils/parallel_algorithms.hpp

```diff
@@ -0,0 +1,132 @@
+/**
+ * traccc library, part of the ACTS project (R&D line)
+ *
+ * (c) 2025 CERN for the benefit of the ACTS project
+ *
+ * Mozilla Public License Version 2.0
+ */
+
+#pragma once
+
+// Local include(s).
+#include "utils.hpp"
+
+// Project include(s).
+#include "traccc/utils/memory_resource.hpp"
+
+// Thrust include(s).
+#include <thrust/binary_search.h>
+#include <thrust/copy.h>
+#include <thrust/execution_policy.h>
+#include <thrust/fill.h>
+#include <thrust/scan.h>
+#include <thrust/sort.h>
+#include <thrust/unique.h>
+
+// OneDPL include(s).
+#if defined(ALPAKA_ACC_SYCL_ENABLED)
+#pragma clang diagnostic push
+#pragma clang diagnostic ignored "-Wshadow"
+#pragma clang diagnostic ignored "-Wunused-parameter"
+#pragma clang diagnostic ignored "-Wshorten-64-to-32"
+#pragma clang diagnostic ignored "-Wsign-conversion"
+#pragma clang diagnostic ignored "-Wimplicit-int-conversion"
+#pragma clang diagnostic ignored "-Wsign-compare"
+#pragma clang diagnostic ignored "-Wold-style-cast"
+#include <oneapi/dpl/algorithm>
+#include <oneapi/dpl/execution>
+#pragma clang diagnostic pop
```

Comment:
You should do something like this instead:

https://github.com/acts-project/traccc/blob/main/device/sycl/src/utils/oneDPL.hpp

Since listing the warnings individually is pretty fragile. (But we do need to turn them off in pragmas unfortunately. 😦)